app/main/forms/suppliers.py

```diff
@@ -28,20 +29,41 @@ class EditSupplierForm(Form):
 class EditContactInformationForm(Form):
     id = IntegerField()
     contactName = StripWhitespaceStringField('Contact name', validators=[
-        DataRequired(message="You must provide a contact name"),
+        InputRequired(message="You must provide a contact name"),
     ])
     email = EmailField('Contact email address', validators=[
-        DataRequired(message="You must provide an email address"),
+        InputRequired(message="You must provide an email address"),
     ])
     phoneNumber = StripWhitespaceStringField('Contact phone number')
+    # TODO: remove these fields when address is removed from the supplier details edit page
     address1 = StripWhitespaceStringField('Building and street')
     city = StripWhitespaceStringField('Town or city')
     postcode = StripWhitespaceStringField('Postcode')
 
 
+class EditRegisteredAddressForm(Form):
+    id = IntegerField()
+    address1 = StripWhitespaceStringField('Building and street', validators=[
+        InputRequired(message="You need to enter the street address."),
+    ])
+    city = StripWhitespaceStringField('Town or city', validators=[
+        InputRequired(message="You need to enter the town or city."),
+    ])
+    postcode = StripWhitespaceStringField('Postcode', validators=[
+        InputRequired(message="You need to enter the postcode."),
+    ])
+
+
+class EditRegisteredCountryForm(Form):
```

Comment:
Why are these separate forms?

Reply:
It's because we're making two separate calls to the apiclient to update the supplier table, as well as update the contact information table. We _could_ do it with one form, but it would involve picking out all the data individually from the form, rather than just passing all the data in as one argument. It was decided it was neater to use separate forms.